### PR TITLE
Fix typos in comments.

### PR DESCRIPTION
**Pull request by ArjunSahlot**
### Description
Fix few typos in the comments.

## Test Plan
No testing required. Only changes comments.

### CODE REVIEW

File: mypy/build.py [hunk 1]
```diff
@@ -1741,7 +1741,7 @@ class State:
     # iteration over dependencies.
     # They should be managed with add_dependency and suppress_dependency.
     dependencies = None  # type: List[str]  # Modules directly imported by the module
-    dependencies_set = None  # type: Set[str]  # The same but as a set for deduplication purposes
+    dependencies_set = None  # type: Set[str]  # The same but as a set for duplication purposes
```
Comment (hauntsaninja) [inline mypy/build.py:1744]: not a typo

File: mypy/checker.py [hunk 2]
```diff
@@ -2188,7 +2188,7 @@ def try_infer_partial_generic_type_from_assignment(self,
             if op != '=' and (typ.type.fullname, op) not in self.partial_type_augmented_ops:
                 return
             # TODO: some logic here duplicates the None partial type counterpart
-            #       inlined in check_assignment(), see # 8043.
+            #       inlined in check_assignment(), see #8043 (https://github.com/python/mypy/issues/8043).
```
Comment (hauntsaninja) [inline mypy/checker.py:2191]: clear from context, i'd undo this change as well

File: mypy/checker.py [hunk 3]
```diff
@@ -4499,15 +4499,15 @@ def refine_identity_comparison_expression(self,
         if target is None:
             return {}, {}
 
-        # If possible, use an unassignable expression as the target.
+        # If possible, use an nonassignable expression as the target.
```
Comment (hauntsaninja) [inline mypy/checker.py:4502]: not a typo

File: mypy/checker.py [hunk 4]
```diff
@@ -4499,15 +4499,15 @@ def refine_identity_comparison_expression(self,
...
         # We skip refining the type of the target below, so ideally we'd
         # want to pick an expression we were going to skip anyways.
         singleton_index = -1
         for i in possible_target_indices:
             if i not in narrowable_operand_indices:
                 singleton_index = i
 
-        # But if none of the possible singletons are unassignable ones, we give up
+        # But if none of the possible singletons are nonassignable ones, we give up
```
Comment (hauntsaninja) [inline mypy/checker.py:4510]: not a typo

File: mypy/checker.py [hunk 5]
```diff
@@ -5477,7 +5477,7 @@ def overload_can_never_match(signature: CallableType, other: CallableType) -> bo
 
 
 def is_more_general_arg_prefix(t: FunctionLike, s: FunctionLike) -> bool:
-    """Does t have wider arguments than s?"""
+    """Does it have wider arguments than s?"""
```
Comment (hauntsaninja) [inline mypy/checker.py:5480]: not a typo

File: mypy/checkexpr.py [hunk 6]
```diff
@@ -90,7 +90,7 @@
 MAX_UNIONS = 5  # type: Final
 
 
-# Types considered safe for comparisons with --strict-equality due to known behaviour of __eq__.
+# Types considered safe for comparisons with --strict-equality due to known behavior of __eq__.
```
Comment (hauntsaninja) [inline mypy/checkexpr.py:93]: mypy seems to use a mix of non-American and American English, so could take or leave this one

**Review by hauntsaninja — CHANGES_REQUESTED** [review]
Welcome to mypy and thanks for your contribution! However many of these typos aren't typos

**Comment by ArjunSahlot** [comment]
Oh thanks for letting me know, this was my first contribution therefore I made multiple mistakes. I have fixed them now though.

**Comment by hauntsaninja** [comment]
Thanks, looks great! :-)
